### PR TITLE
16 add tests on contributors in few classes simultanuosly

### LINKED ISSUE #16: Chore: Add tests on contributors in few classes simultanuosly
## Status Quo

Currently we have few different classes, but no test of the seed contract functioning with few user classes.

## Acceptance criteria
- Test full contract functionality with multiple user classes for whitelisted version(Creation, adding class, changing class, whitelisting users, changing class of users, unwhitelisting users, buying Seed, withdrawing funds back, succeeding/not succeeding in funds gathering, seed withdrawal, claiming funds back).
- Test full contract functionality with multiple user classes for not-whitelisted version.

### PR DESCRIPTION
**Pull request by decadanse**
Current whitelisted and non-whitelisted tests moved down to the end of the file for the purpose of removing only-there-used time variables and increases of time. 

NOTE:  
CLASS_FEE = parseEther("0.21").toString(); //21% 
require < 45%
BUT when it's 0.22 or more - reverted with panic code 0x11 (Arithmetic operation underflowed or overflowed outside of an unchecked block) appears

UPD:
The issue, which is fixes this 0x11 error was created. 

closes #16

### CODE REVIEW

**Review by arseneeth — APPROVED** [review]

File: contracts/seed/Seed.sol [hunk 1]
```diff
@@ -21,7 +21,7 @@ pragma solidity 0.8.9;
 
 import "openzeppelin-contracts-sol8/token/ERC20/IERC20.sol";
 import "openzeppelin-contracts-sol8/token/ERC20/utils/SafeERC20.sol";
-
+import "hardhat/console.sol";
```
Comment (BritikovKI) [inline contracts/seed/Seed.sol:24]: Remove this pls

File: test/seed.js [hunk 2]
```diff
@@ -2668,4 +2672,805 @@ describe("Contract: Seed", async () => {
...
+            claimAmount_class2 = new BN(time.duration.days(1)).mul(
+                new BN(buySeedAmount)
+                    .mul(new BN(PRECISION.toString()))
+                    .div(new BN(price))
+                    .div(new BN(localVestingDuration)));
+            feeAmount_class2 = new BN(claimAmount_class1)
+                .mul(new BN(SECOND_CLASS_FEE))
+                .div(new BN(PRECISION.toString()));
+        });
+        it("adds new classes", async () => {
```
[the remark this answers is not in the document]
Comment (decadanse) [reply test/seed.js:2813]: done

File: test/seed.js [hunk 3]
```diff
@@ -2668,4 +2672,805 @@ describe("Contract: Seed", async () => {
...
+              .connect(admin)
+              .changeClass(1, CLASS_25_PERSONAL_FUNDING_LIMIT, CLASS_20_PERSONAL_FUNDING_LIMIT, price_class1, localVestingDuration.toNumber(), newClassVestingStartTime.toNumber(), CLASS_FEE);
+            expect(
+              (await setup.seed.getClass(1))[1]
+            ).to.equal((ethers.BigNumber.from(CLASS_20_PERSONAL_FUNDING_LIMIT)));
+            expect(
+                (await setup.seed.getClass(1))[2]
+            ).to.equal((ethers.BigNumber.from(price_class1)));
+        });
+        it("it sets class", async () => {
```
[the remark this answers is not in the document]
Comment (decadanse) [reply test/seed.js]: fixed
"it whitelists users" now

File: contracts/seed/Seed.sol (hunk 1 above)
Comment (decadanse) [reply contracts/seed/Seed.sol]: removed

**Pull request merged**